src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/poptimizer/simple/DPBasedLinearJoinPlanOptimizer.java

```diff
@@ -0,0 +1,33 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.poptimizer.simple;
+
+import se.liu.ida.hefquin.engine.queryproc.impl.poptimizer.QueryOptimizationContext;
+import se.liu.ida.hefquin.engine.utils.Pair;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class DPBasedLinearJoinPlanOptimizer extends DPBasedJoinPlanOptimizer {
+
+    public DPBasedLinearJoinPlanOptimizer( final QueryOptimizationContext ctxt ) {
+        super(ctxt);
+    }
+
+    @Override
+    public <T> List<Pair<List<T>, List<T>>> splitIntoSubSets( final List<T> superset ) {
```

Comment:
Same request for this one. Please create a unit test with `List<Integer>` containing three elements.